syntax = "proto3";

package jacobe.twirpy_experiment.test;
import "google/protobuf/empty.proto";
import "google/protobuf/field_mask.proto";

// Generated according to https://cloud.google.com/apis/design/standard_methods
service TestService {
  rpc ListTests(ListTestsRequest) returns (ListTestsResponse);

  rpc GetTest(GetTestRequest) returns (Test);

  rpc CreateTest(CreateTestRequest) returns (Test);

  rpc UpdateTest(UpdateTestRequest) returns (Test);

  rpc DeleteTest(DeleteTestRequest) returns (google.protobuf.Empty);
}

message Test {
  string name = 1;
  string description = 2;
  int64 id = 3;
  string created_at = 4;
  string updated_at = 5;
}

message ListTestsRequest {
  // The parent resource name, for example, "shelves/shelf1"
  string parent = 1;

  // The maximum number of items to return.
  int32 page_size = 2;

  // The next_page_token value returned from a previous List request, if any.
  string page_token = 3;
}

message ListTestsResponse {
  // The field name should match the noun "Test" in the method name.
  // There will be a maximum number of items returned based on the page_size field in the request.
  repeated Test tests = 1;

  // Token to retrieve the next page of results, or empty if there are no more results in the list.
  string next_page_token = 2;
}

message GetTestRequest {
  // The field will contain name of the resource requested.
  string name = 1;
}

message CreateTestRequest {
  // The parent resource name where the Test is to be created.
  string parent = 1;

  // The Test id to use for this Test.
  string test_id = 2;

  // The Test resource to create.
  // The field name should match the Noun in the method name.
  Test test = 3;
}

message UpdateTestRequest {
  // The Test resource which replaces the resource on the server.
  Test test = 1;

  // The update mask applies to the resource. For the `FieldMask` definition,
  // see https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#fieldmask
  google.protobuf.FieldMask update_mask = 2;
}

message DeleteTestRequest {
  // The resource name of the Test to be deleted.
  string name = 1;
}
